presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -640,7 +640,7 @@ protected Scope visitCreateTableAsSelect(CreateTableAsSelect node, Optional<Scop
             else {
                 validateColumns(node, queryScope.getRelationType());
             }
-
+            validateBaseTables(analysis.getTableNodes(), node);
```

Comment:
Let's try to put the similar validation logic at the same place in visitTable() line 1220, probably do analysis.getStatement() instanceof CreateTableAsSelect.